src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +23,95 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
     {
         private readonly ITagAggregator<InlineParameterNameHintDataTag> _tagAggregator;
         private readonly ITextBuffer _buffer;
+        private readonly ITextView _textView;
+        private readonly List<ITagSpan<IntraTextAdornmentTag>> _cache;
+        private ITextSnapshot _cacheSnapshot;
+        private readonly IClassificationFormatMap _formatMap;
+        private TextFormattingRunProperties _format;
+        private readonly IClassificationType _keyword;
 
         public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
 
-        public InlineParameterNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
+        public InlineParameterNameHintsTagger(InlineParameterNameHintsTaggerProvider taggerProvider, ITextView textView, ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
         {
+            _cache = new List<ITagSpan<IntraTextAdornmentTag>>();
+            _textView = textView;
             _buffer = buffer;
             _tagAggregator = tagAggregator;
+            _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
+            _keyword = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineParameterNameHintsTag.TagId);
+            _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
             _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
         }
 
+        private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
```

Comment:
This only runs on the UI thread?

Reply:
The assembly only says this ```Microsoft.VisualStudio.Text.UI.Wpf.dll```, so I believe so

Reply:
we have ways of asserting this.  it woudl be good for us to do that assertion so that if this ever turns out to be a wrong assumption we can get a crash immediately instead of a hard to figure out race condition.